### DISCUSSION TITLE
Composing components while adding extra slots

### DISCUSSION THREAD

**Discussion by lewebsimple** [heroui-inc/tailwind-variants#21] [Q&A] [answered]
As the title says, is it possible to add extra slots when extending a component ?

Consider the following:
```ts
import { tv } from "tailwind-variants";

const section = tv({
  slots: {
    wrapper: '',
    inner: '',
  },
});

const extended = tv({
  extend: section,
  slots: {
    inner: 'py-12',
    extraSlot: '',
  }
});

// This doesn't work as extended().extraSlot is undefined, but commenting out `extend: section` makes it work
console.log(extended().extraSlot())
```

Of course adding the missing slots in the parent component with empty values would fix the problem, but this means we have to know every possible slot name in advance ... which is not very convenient DX or downright impossible for packaged libraries.

Am I missing something ?

**Comment by jrgarciadev** [comment, marked as answer]
Hey @lewebsimple thank you for reporting it, I just sent a PR to fix it https://github.com/nextui-org/tailwind-variants/pull/22

**Reply by lewebsimple** [reply]
This is an incredibly fast response plus the PR adds other features as well, thanks!!

**Reply by jrgarciadev** [reply]
Please upgrade to the v0.1.0 🚀
